space-cli/cmd/modules/auth/commands.go

```diff
@@ -81,3 +81,46 @@ func actionGenerateUserManagement(cmd *cobra.Command, args []string) error {
 
 	return utils.AppendConfigToDisk(dbrule, dbruleConfigFile)
 }
+
+// DeleteSubCommands is the list of commands the auth module exposes
+func DeleteSubCommands() []*cobra.Command {
+
+	var deleteAuthProvider = &cobra.Command{
+		Use:     "auth-provider",
+		Aliases: []string{"auth-providers"},
+		RunE:    actionDeleteAuthProvider,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			project, check := utils.GetProjectID()
+			if !check {
+				utils.LogDebug("Project not specified in flag", nil)
+				return nil, cobra.ShellCompDirectiveDefault
+			}
+			objs, err := GetAuthProviders(project, "auth-providers", map[string]string{})
+			if err != nil {
+				return nil, cobra.ShellCompDirectiveDefault
+			}
+			var ids []string
+			for _, v := range objs {
+				ids = append(ids, v.Meta["id"])
+			}
+			return ids, cobra.ShellCompDirectiveDefault
+		},
```

Comment:
I think this piece of code is also written in get auth-provider command.
I suggest you make a file called helpers.go & define this function in that file & use this function in getter & delete subcommand.
Do the same for other commands also, DB-rules, DB-config etc...
Keep the code DRY